PhysicsTools/PatUtils/plugins/L1PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,469 @@
+        }
+        //If overlapping photons have a non prefiring rate larger than the jet, then replace these weights by the jet one
+        else if (nonprefiringprobfromoverlappingphotons > nonprefiringprobfromoverlappingjet) {
+          if (nonprefiringprobfromoverlappingphotons > 0.) {
+            nonPrefiringProba[fluct] *= nonprefiringprobfromoverlappingjet / nonprefiringprobfromoverlappingphotons;
+            nonPrefiringProbaJet[fluct] *= nonprefiringprobfromoverlappingjet / nonprefiringprobfromoverlappingphotons;
+          } else {
+            nonPrefiringProba[fluct] = 0.;
+            nonPrefiringProbaJet[fluct] = 0.;
+          }
+        }
+        //Last case: if overlapping photons have a non prefiring rate smaller than the jet, don't consider the jet in the event weight, and do nothing.
+      }
+    }
+    //Now calculate prefiring weights for muons
+    if (!missingInputMuon_) {
+      for (const auto& muon : theMuons) {
+        double pt = muon.pt();
+        double phi = muon.phi();
+        double eta = muon.eta();
+        // Remove crappy tracker muons which would not have prefired the L1 trigger
+        if (pt < 5 && !muon.isStandAloneMuon())
+          continue;
+        double prefiringprob_mu = getPrefiringRateMuon(eta, phi, pt, fluct);
+        nonPrefiringProba[fluct] *= (1. - prefiringprob_mu);
+        nonPrefiringProbaMuon[fluct] *= (1. - prefiringprob_mu);
+      }
+    }
+  }
+  for (const auto fluct :
+       {fluctuations::upSyst, fluctuations::downSyst, fluctuations::upStat, fluctuations::downStat}) {
+    if (!missingInputMuon_) {
+      for (const auto& muon : theMuons) {
+        double pt = muon.pt();
+        double phi = muon.phi();
+        double eta = muon.eta();
+        // Remove crappy tracker muons which would not have prefired the L1 trigger
+        if (pt < 5 && !muon.isStandAloneMuon())
+          continue;
+        double prefiringprob_mu = getPrefiringRateMuon(eta, phi, pt, fluct);
+        nonPrefiringProbaMuon[fluct] *= (1. - prefiringprob_mu);
+      }
+    }
+  }
+  //Move global prefire weights, as well as those for muons, photons, and jets, to the event
+  iEvent.emplace(nonPrefiringProbToken_, nonPrefiringProba[0]);
+  iEvent.emplace(nonPrefiringProbUpToken_, nonPrefiringProba[1]);
+  iEvent.emplace(nonPrefiringProbDownToken_, nonPrefiringProba[2]);
+
+  iEvent.emplace(nonPrefiringProbJetToken_, nonPrefiringProbaJet[0]);
+  iEvent.emplace(nonPrefiringProbJetUpToken_, nonPrefiringProbaJet[1]);
+  iEvent.emplace(nonPrefiringProbJetDownToken_, nonPrefiringProbaJet[2]);
+
+  iEvent.emplace(nonPrefiringProbPhotonToken_, nonPrefiringProbaPhoton[0]);
+  iEvent.emplace(nonPrefiringProbPhotonUpToken_, nonPrefiringProbaPhoton[1]);
+  iEvent.emplace(nonPrefiringProbPhotonDownToken_, nonPrefiringProbaPhoton[2]);
+
+  iEvent.emplace(nonPrefiringProbMuonToken_, nonPrefiringProbaMuon[0]);
+  iEvent.emplace(nonPrefiringProbMuonUpSystToken_, nonPrefiringProbaMuon[3]);
```

Comment:
[1] , [2] are missing here...
Ok, my comments are somehow touching what you are updating.... Let us know when you are done with the changes

Reply:
That was intentional, 1 and 2 are the full uncertainty, 3-6 are split into stat. and syst, which is requested for muons by xpog. I could add 1 and 2 to the event as well, even though it will not be picked up by nanoAOD. But it is true that other users might find it convenient to have the total uncertainty available.

Reply:
I have restored the combined uncertainty to the output of the producer.

Reply:
@JanFSchulte  Looks to me that [3] ([4]) is swapped with [5] ([6]) ?

Reply:
Yes, you are right! I'll fix that asap.